src/main/java/org/spongepowered/api/item/recipe/smelting/SmeltingRegistry.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.recipe.smelting;
+
+import org.spongepowered.api.item.ItemType;
+import org.spongepowered.api.item.inventory.ItemStack;
+import org.spongepowered.api.item.inventory.ItemStackSnapshot;
+import org.spongepowered.api.item.recipe.RecipeRegistry;
+
+import java.util.Optional;
+
+/**
+ * A SmeltingRegistry holds all registered smelting recipes for a given game.
+ */
+public interface SmeltingRegistry extends RecipeRegistry<SmeltingRecipe> {
+
+    /**
+     * Gets the experience a {@link ItemType} drops when it is made.
+     *
+     * @param stack the {@link ItemStack}
+     * @return the experience of the {@link ItemStack}
+     */
+    double getExperience(ItemStackSnapshot stack);
+
+    /**
+     * Returns the results for running this SmeltingRegistry over an {@link ItemType}.
+     *
+     * @param type the type to search for
+     * @return A {@link ItemStackSnapshot} or {@link Optional#empty()} if the given
+     *          {@link ItemType} does not match any recipes.
+     */
+    Optional<ItemStackSnapshot> getResult(ItemType type);
```

Comment:
I put ItemType there, because it ignores stack size. I'll probably be changing getExperience to that too